milestones/api.py

```diff
@@ -77,12 +77,19 @@ def get_milestone_relationship_types():
     return data.fetch_milestone_relationship_types()
 
 
-def add_milestone(milestone):
+def add_milestone(milestone, propagate=True):
     """
     Passes a new milestone to the data layer for storage
+
+    Arguments:
+        milestone (dict): The milestone to persist
+        propagate (bool): False to prevent reactivation of soft-deleted milestone relationships
```

Comment:
Can you elaborate on the use case for when we would want to reactivate a pre-existing milestone but would not want to reactivate all of the corresponding relationships for that milestone?